Add configurable time format support

mattrobenolt opened the pull request:
- Add time_format config option: 'detect', '12h', '24h'
- Auto-detects system preference on macOS, defaults to 12h elsewhere
- Updates TUI to display timestamps in user's preferred format

mattrobenolt committed:
Add configurable time format support

- Add time_format config option: 'detect', '12h', '24h'
- Auto-detects system preference on macOS, defaults to 12h elsewhere
- Updates TUI to display timestamps in user's preferred format

adamdotdevin commented on packages/tui/internal/components/chat/message.go:
```diff
@@ -22,6 +24,38 @@ import (
+		return true
+	case "12h":
+		return false
+	default:
+		// Default to detect behavior
+		if runtime.GOOS == "darwin" {
+			return detectMacOS24HourFormat()
+		}
+		return false // fallback to 12h on other platforms
```
i like the UX of it auto-determining on mac os, but with the way bubbletea works this code is going to run a lot; we should either 1) drop the auto-detect and just default to 12h unless you configure or 2) store the detected format in `app.State` and use it if there's no configured value.

mattrobenolt replied:
Done, I think this is what you're more expecting. I'd typically use a sync.OnceValue or something here, but I don't think there's anything that could happen as a race condition, so just used a *bool on the app.State.

mattrobenolt committed:
Use lazy evaluation for time format detection

Cache system detection result in app state to avoid running date command on every message render

adamdotdevin commented:
only thing i see missing is a call to app.SaveState to save the detected time format!
